### PR TITLE
Fixed issue #2083

### PR DESCRIPTION
**Pull request by bumbadadabum**
Or I tried, at least.

### CODE REVIEW

File: battle-engine.js [hunk 1]
```diff
@@ -336,6 +336,7 @@ BattlePokemon = (function () {
 	BattlePokemon.prototype.hpType = 'Dark';
 	BattlePokemon.prototype.hpPower = 60;
 	BattlePokemon.prototype.speed = 0;
+	BattlePokemon.prototype.turnSwitchedIn = 0;
```
Comment (Zarel) [inline battle-engine.js]: You can calculate this from `battle.turn - pokemon.activeTurns`, can't you?

Comment (bumbadadabum) [inline battle-engine.js]: I can.

**Comment by Zarel** [comment]
@Marty-D is it sorted by the number of turns a Pokémon is out? Or is it, as I suspect, more like "the order the Pokémon were switched in"?

**Comment by bumbadadabum** [comment]
I miiiiight have misinterpreted then.

**Comment by ascriptmaster** [comment]
I believe it's the order the Pokemon were switched in, because I remember Marty mentioning that the leftmost (or was it rightmost?) Pokemon is used in the case of simultaneous switches, but I don't remember the order Pokemon get switched in in those cases.

**Comment by Marty-D** [comment]
Alright, so since my test with V4 never happened, I thought up a way to test this alone by abusing Transform mechanics. By Skill Swapping Lightning Rod to an opposing Magikarp, then switching Imposter Ditto into the slot facing it, I get two Speed tying Lightning Rod Pokemon on opposite sides.

Doing the test in that order makes the Magikarp take all the (single-target) Electric moves aimed anywhere. So next, I Transformed into it first, then Skill Swapped Lightning Rod onto my Ditto, then Skill Swapped another Lightning Rod onto Magikarp. Now Ditto takes all the Electric moves.

This is consistent with my Gen 5 tests, except it's clear now that redirection Speed ties aren't determined by which Pokemon has been on the field longest, but by which Pokemon has had Lightning Rod the longest (which is usually the same thing if Ability changes aren't in play).

This probably also means that the person hosting the battle determines which side gets the advantage on Turn 1. Player 2 (not the host) always sends out their Pokemon first, therefore they have their Abilities active longer by default. I'd like to actually test this part sometime though. ****cough**** @DanUgore

**Comment by Joimer** [comment]
One more thing, maybe you can add an automated test for this, would make it easier to check if a change complies with all mechanics.

**Comment by Slayer95** [comment]
@ascriptmaster has written some tests in https://github.com/Zarel/Pokemon-Showdown/pull/2130

**Comment by Joimer** [comment]
Oh, both go kind of together, don't they? Even though he said he doesn't care anymore for the PR. I guess both should be merged into one.

**Pull request closed without merging**